edgexfoundry/app-functions-sdk-go
# feat: Consume new -d/--dev Dev Mode command-line flag

Dev Mode is used when running service locally with the other services running in Docker (aka hybrid mode). This causes all the Host settings pulled from common configuration to be overridden with the value localhost.







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

https://github.com/edgexfoundry/edgex-docs/pull/1040 

## Testing Instructions
From latest Compose Builder run  make run no-secty ds-virtual
run `make build` from this branch
run the app template with just `-cp` flag
Verify the following error
```
"Unable to get version of Core Metadata: failed to send a http request -> Get \"http://edgex-core-metadata:59881/api/v3/version\": dial tcp: lookup edgex-core-metadata on 8.8.8.8:53: no such host"
```
run the app template with just `-cp -d` flags
Verify no errors
Enable DEBUG logging from Consul
Verify service is receiving and processing events

## New Dependency Instructions (If applicable)
